apps/website-25/src/components/courses/exercises/FreeTextResponse.tsx

```diff
@@ -1,36 +1,55 @@
 import clsx from 'clsx';
-import CTALinkOrButton from '@bluedot/ui/src/CTALinkOrButton';
-import React from 'react';
+import { CTALinkOrButton } from '@bluedot/ui';
+import React, { useCallback, useEffect } from 'react';
 import ReactMarkdown from 'react-markdown';
+import { useForm } from 'react-hook-form';
 
 type FreeTextResponseProps = {
   // Required
-  title: string;
-  description: string;
   className?: string;
+  description: string;
+  onExerciseSubmit: (exerciseResponse: string) => void;
+  title: string;
+  // Optional
+  exerciseResponse?: string;
+};
+
+type FormData = {
+  answer: string;
 };
 
 const FreeTextResponse: React.FC<FreeTextResponseProps> = ({
-  title,
-  description,
   className,
+  description,
+  exerciseResponse,
+  onExerciseSubmit,
+  title,
 }) => {
-  const [submittedOption, setSubmittedOption] = React.useState<string | null>(null);
-  const [isSaved, setIsSaved] = React.useState<boolean>(false);
+  const [isEditing, setIsEditing] = React.useState<boolean>(false);
+  const { register, handleSubmit, setValue } = useForm<FormData>({
+    defaultValues: {
+      answer: exerciseResponse || '',
+    },
+  });
 
-  const handleAnswerSubmit = () => {
-    setSubmittedOption('test');
-    setIsSaved(true);
-  };
+  useEffect(() => {
+    if (exerciseResponse) {
+      setValue('answer', exerciseResponse);
+    }
+  }, [exerciseResponse, setValue]);
 
-  const handleAnswerChange = () => {
-    if (submittedOption && isSaved) {
-      setIsSaved(false);
+  const onSubmit = useCallback(async (data: FormData) => {
+    try {
+      await onExerciseSubmit(data.answer);
+      setIsEditing(false);
+    } catch (e) {
+      console.log('error', e);
+      // Set some state for the error and render it
```

Comment:
nit:

```suggestion
      // TODO: Set some state for the error and render it
```

just so it's easier to find and bash these later